### PR TITLE
[cxxmodules] Do not use FindObject when we have the Decl*.

### PR DESCRIPTION
**Pull request by vgvassilev**

### CODE REVIEW

**Comment by vgvassilev** [comment]
@phsft-bot build!

File: core/metacling/src/TCling.cxx [hunk 1]
```diff
@@ -555,15 +555,16 @@ void TCling::HandleNewDecl(const void* DV, bool isDeserialized, std::set<TClass*
       // scope.
       if (!(isa<VarDecl>(ND)))
          return;
-
+      auto ListOfGlobals = static_cast<TListOfDataMembers*>(gROOT->GetListOfGlobals());
+      const VarDecl *CanonVD = cast<VarDecl>(ND)->getCanonicalDecl();
       // Skip if already in the list.
-      if (gROOT->GetListOfGlobals()->FindObject(ND->getNameAsString().c_str()))
+      if (ListOfGlobals->Find(static_cast<DeclId_t>(CanonVD)))
```
Comment (pcanal) [inline core/metacling/src/TCling.cxx]: Would be it be better as a const_cast and/or an update to TListOfDataMembers::Find to take a const decl?

Comment (vgvassilev) [reply core/metacling/src/TCling.cxx]: I am not sure I understand the `const_cast` comment. I agree we may want to update `TListOfDataMembers::Find` as a next step in improving this code.

Comment (pcanal) [reply core/metacling/src/TCling.cxx]: My bad. I misread the line number in the error messages.

**Comment by Axel-Naumann** [comment]
The behavior shown in the failing test is not what we need, because the old `TGlobal` object claims it's valid but finding it by name returns a new one. I would not mind discussing whether returning a new `TGlobal` object from `GetListOfGlobals()->FindObject()` is a worthwhile backward incompatible change, but then the old `TGlobal` object must be told that it has been invalidated.

Do you want to update the old `TGlobal` object and start that discussion? Or do you have a plan B?

**Pull request merged**